experiments/targeting.go

```diff
@@ -0,0 +1,298 @@
+	values []interface{}
+}
+
+func NewEqualNode(inputNodes map[string]interface{}) (Targeting, error) {
+	if len(inputNodes) != 2 {
+		return nil, TargetingNodeError("EqualNode expects exactly two fields")
+	}
+	acceptedKey, ok := inputNodes["field"]
+	if !ok {
+		return nil, TargetingNodeError("EqualNode expects input key 'field'")
+	}
+
+	value, valueOK := inputNodes["value"]
+	values, valuesOK := inputNodes["values"]
+	if !valueOK && !valuesOK {
+		return nil, TargetingNodeError("EqualNode expects input key 'value' or 'values'.")
+	}
+
+	var acceptedValues []interface{}
+	if valueOK {
+		acceptedValues = []interface{}{value}
+	}
+	if valuesOK {
+		acceptedValues = values.([]interface{})
+	}
+	return &EqualNode{
+		field:  strings.ToLower(acceptedKey.(string)),
+		values: acceptedValues,
+	}, nil
+}
+
+func (n *EqualNode) Evaluate(inputs map[string]interface{}) bool {
+	candidateValue := inputs[n.field]
+	switch n := candidateValue.(type) {
+	case int, float64:
+		candidateValue = json.Number(fmt.Sprintf("%v", n))
+	}
+	for _, value := range n.values {
+		if candidateValue == value {
+			return true
+		}
+	}
+	return false
+}
+
+type NotNode struct {
+	child Targeting
+}
+
+func NewNotNode(inputNodes map[string]interface{}) (*NotNode, error) {
+	if len(inputNodes) != 1 {
+		return nil, TargetingNodeError("NotNode expects exactly one field")
+	}
+	node, err := parseNode(inputNodes)
+	if err != nil {
+		return nil, err
+	}
+	return &NotNode{
+		child: node,
+	}, nil
+}
+
+func (n *NotNode) Evaluate(inputs map[string]interface{}) bool {
+	return !n.child.Evaluate(inputs)
+}
+
+type OverrideNode struct {
+	ReturnValue bool
+}
+
+func NewOverrideNode(inputNode interface{}) *OverrideNode {
+	returnValue, ok := inputNode.(bool)
+	if !ok {
+		returnValue = false
+	}
+	return &OverrideNode{
+		ReturnValue: returnValue,
+	}
+}
+
+func (n *OverrideNode) Evaluate(inputs map[string]interface{}) bool {
+	return n.ReturnValue
+}
+
+type ComparisonNode struct {
+	field string
+	value interface{}
+	less  Less
+}
+
+func NewComparisonNode(inputs map[string]interface{}, less Less) (*ComparisonNode, error) {
+	if len(inputs) != 2 {
+		return nil, TargetingNodeError("ComparisonNode expects exactly two fields")
+	}
+	field, ok := inputs["field"]
+	if !ok {
+		return nil, TargetingNodeError("ComparisonNode expects input key 'field'.")
+	}
+	value, ok := inputs["value"]
+	if !ok {
+		return nil, TargetingNodeError("ComparisonNode expects input key 'value'.")
+	}
+	return &ComparisonNode{
+		field: field.(string),
+		value: value,
+		less:  less,
+	}, nil
+}
+
+func (n *ComparisonNode) Evaluate(inputs map[string]interface{}) bool {
+	candidateValue := inputs[n.field]
+	if n.value == nil {
+		return false
+	}
+	value, err := (n.value.(json.Number)).Float64()
+	if err != nil {
+		return false
+	}
+	switch cv := candidateValue.(type) {
+	case int:
+		return n.less(float64(cv), value)
+	case float64:
+		n.less(cv, value)
+	}
+	return false
+}
+
+type Targeting interface {
```

Comment:
style nit: can you move this to the top of the file so it's easier to tell that the `Node`s implement this?